packages/dataclass-editor/src/components/dataclass/detail/FieldDetailContent.tsx

```diff
@@ -1,20 +1,33 @@
-import { BasicCheckbox, BasicField, Flex, Input, Textarea } from '@axonivy/ui-components';
+import {
+  BasicCheckbox,
+  BasicField,
+  Collapsible,
+  CollapsibleContent,
+  CollapsibleTrigger,
+  Flex,
+  Input,
+  Textarea
+} from '@axonivy/ui-components';
+import { useEffect, useState } from 'react';
 import { useAppContext } from '../../../context/AppContext';
 import { removeEmptyStrings } from '../../../utils/array/array';
 import type { DataClassField } from '../data/dataclass';
 import './DetailContent.css';
 
 export const FieldDetailContent = () => {
   const { dataClass, setDataClass, selectedField } = useAppContext();
-  if (selectedField === undefined || dataClass.fields.length - 1 < selectedField) {
+  const field = selectedField !== undefined && selectedField < dataClass.fields.length ? dataClass.fields[selectedField] : undefined;
+
+  const [annotationsOpen, setAnnotationsOpen] = useState(false);
+  useEffect(() => setAnnotationsOpen(field ? field.annotations.length !== 0 : false), [field]);
```

Comment:
Do not understand why you need this state and this useEffect...
Why do you not simple use the defaultOpen prop with `field.annotations.length > 0

Reply:
I tried but when having a field selected and then selecting another field the open state of this collapsible does not get updated.

As I understand, the `defaultOpen` property only controls the initial state when the component is first mounted and will not re-render if the value changes.

E.g: First, I select a field without annotations so the annotations collapsible will be closed. Next, I select another field with annotations so the annotations collapsible should be open. However, it will stay closed.

Reply:
Ah yes your right.
Technically there is a trick to let it rerender. It should work if you set a key prop to the collapsible with e.g. the fieldname as value. 
But your way is also fine 👍

Reply:
Thanks for the tip. I think that's much cleaner. Also because I don't have to hold the state for all the other fields which I am putting into a collapsible now :smile: